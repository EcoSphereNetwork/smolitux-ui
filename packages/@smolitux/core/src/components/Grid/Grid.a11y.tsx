import React, { forwardRef } from 'react';
import { Grid, GridProps } from './Grid';

export interface GridA11yProps extends GridProps {
  /**
   * ARIA-Label für das Grid
   */
  ariaLabel?: string;
  
  /**
   * ARIA-Labelledby für das Grid
   */
  ariaLabelledby?: string;
  
  /**
   * ARIA-Describedby für das Grid
   */
  ariaDescribedby?: string;
  
  /**
   * ARIA-Rolle für das Grid
   */
  role?: string;
  
  /**
   * Ob das Grid eine Region ist
   */
  isRegion?: boolean;
  
  /**
   * Ob das Grid eine Landmark ist
   */
  isLandmark?: boolean;
  
  /**
   * Ob das Grid eine Tabelle ist
   */
  isTable?: boolean;
  
  /**
   * Ob das Grid ein Formular ist
   */
  isForm?: boolean;
  
  /**
   * Ob das Grid eine Navigation ist
   */
  isNavigation?: boolean;
  
  /**
   * Ob das Grid eine Liste ist
   */
  isList?: boolean;
  
  /**
   * Ob das Grid ein Menü ist
   */
  isMenu?: boolean;
  
  /**
   * Ob das Grid ein Dialog ist
   */
  isDialog?: boolean;
  
  /**
   * Ob das Grid ein Alert ist
   */
  isAlert?: boolean;
  
  /**
   * Ob das Grid ein Status ist
   */
  isStatus?: boolean;
  
  /**
   * Ob das Grid ein Live-Region ist
   */
  isLiveRegion?: boolean;
  
  /**
   * Politeness-Level für Live-Region
   */
  liveRegionPoliteness?: 'polite' | 'assertive' | 'off';
  
  /**
   * Ob das Grid atomar ist
   */
  isAtomic?: boolean;
  
  /**
   * Ob das Grid relevant ist
   */
  isRelevant?: boolean;
  
  /**
   * Relevanz-Typ für das Grid
   */
  relevantType?: 'additions' | 'removals' | 'text' | 'all';
  
  /**
   * Ob das Grid busy ist
   */
  isBusy?: boolean;
  
  /**
   * Ob das Grid fokussierbar ist
   */
  isFocusable?: boolean;
  
  /**
   * Tab-Index für das Grid
   */
  tabIndex?: number;
  
  /**
   * Ob das Grid eine Tabelle ist
   */
  isTable2?: boolean;
  
  /**
   * Ob das Grid eine Tabellen-Kopfzeile hat
   */
  hasTableHeader?: boolean;
  
  /**
   * Ob das Grid eine Tabellen-Fußzeile hat
   */
  hasTableFooter?: boolean;
  
  /**
   * Ob das Grid eine Tabellen-Beschreibung hat
   */
  hasTableCaption?: boolean;
  
  /**
   * Tabellen-Beschreibung
   */
  tableCaption?: string;
  
  /**
   * Ob das Grid eine Tabellen-Zusammenfassung hat
   */
  hasTableSummary?: boolean;
  
  /**
   * Tabellen-Zusammenfassung
   */
  tableSummary?: string;
}

/**
 * Barrierefreie Grid-Komponente für Layouts
 * 
 * @example
 * ```tsx
 * <GridA11y columns={2} gap="md" ariaLabel="Beispiel-Grid">
 *   <div>Element 1</div>
 *   <div>Element 2</div>
 * </GridA11y>
 * ```
 */
export const GridA11y = forwardRef<HTMLDivElement, GridA11yProps>(({
  ariaLabel,
  ariaLabelledby,
  ariaDescribedby,
  role,
  isRegion = false,
  isLandmark = false,
  isTable = false,
  isForm = false,
  isNavigation = false,
  isList = false,
  isMenu = false,
  isDialog = false,
  isAlert = false,
  isStatus = false,
  isLiveRegion = false,
  liveRegionPoliteness = 'polite',
  isAtomic = false,
  isRelevant = false,
  relevantType = 'additions',
  isBusy = false,
  isFocusable = false,
  tabIndex,
  isTable2 = false,
  hasTableHeader = false,
  hasTableFooter = false,
  hasTableCaption = false,
  tableCaption,
  hasTableSummary = false,
  tableSummary,
  ...props
}, ref) => {
  // Bestimme die Rolle basierend auf den Eigenschaften
  const determineRole = () => {
    if (role) return role;
    if (isRegion) return 'region';
    if (isLandmark) return 'landmark';
    if (isTable || isTable2) return 'table';
    if (isForm) return 'form';
    if (isNavigation) return 'navigation';
    if (isList) return 'list';
    if (isMenu) return 'menu';
    if (isDialog) return 'dialog';
    if (isAlert) return 'alert';
    if (isStatus) return 'status';
    return undefined;
  };
  
  // Bestimme die ARIA-Live-Region-Eigenschaften
  const determineAriaLive = () => {
    if (!isLiveRegion) return undefined;
    return liveRegionPoliteness;
  };
  
  // Bestimme die ARIA-Relevant-Eigenschaft
  const determineAriaRelevant = () => {
    if (!isRelevant) return undefined;
    return relevantType;
  };
  
  // Rendere das Grid mit den entsprechenden ARIA-Attributen
  return (
    <>
      {/* Tabellen-Beschreibung */}
      {(isTable || isTable2) && hasTableCaption && tableCaption && (
        <div className="sr-only" id={`${props.id || 'grid'}-caption`}>
          {tableCaption}
        </div>
      )}
      
      {/* Tabellen-Zusammenfassung */}
      {(isTable || isTable2) && hasTableSummary && tableSummary && (
        <div className="sr-only" id={`${props.id || 'grid'}-summary`}>
          {tableSummary}
        </div>
      )}
      
      <Grid
        ref={ref}
        role={determineRole()}
        aria-label={ariaLabel}
        aria-labelledby={ariaLabelledby || ((isTable || isTable2) && hasTableCaption ? `${props.id || 'grid'}-caption` : undefined)}
        aria-describedby={ariaDescribedby || ((isTable || isTable2) && hasTableSummary ? `${props.id || 'grid'}-summary` : undefined)}
        aria-live={determineAriaLive()}
        aria-atomic={isLiveRegion && isAtomic ? true : undefined}
        aria-relevant={determineAriaRelevant()}
        aria-busy={isBusy ? true : undefined}
        tabIndex={isFocusable ? (tabIndex || 0) : undefined}
        {...props}
      />
    </>
  );
});

GridA11y.displayName = 'GridA11y';

export default GridA11y;