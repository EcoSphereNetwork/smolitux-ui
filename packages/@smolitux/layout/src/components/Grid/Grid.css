.smolitux-grid {
  display: grid;
  width: 100%;
  box-sizing: border-box;
}

.smolitux-grid-item {
  box-sizing: border-box;
}

/* Spacing-Klassen */
.smolitux-grid.spacing-none {
  gap: 0;
}

.smolitux-grid.spacing-xs {
  gap: 0.25rem;
}

.smolitux-grid.spacing-sm {
  gap: 0.5rem;
}

.smolitux-grid.spacing-md {
  gap: 1rem;
}

.smolitux-grid.spacing-lg {
  gap: 1.5rem;
}

.smolitux-grid.spacing-xl {
  gap: 2rem;
}

/* Grid-Spalten */
.smolitux-grid {
  grid-template-columns: repeat(12, 1fr);
}

/* Responsive Grid-Items */
@media (max-width: 576px) {
  .smolitux-grid-item[data-xs="1"] { grid-column: span 1 / span 1; }
  .smolitux-grid-item[data-xs="2"] { grid-column: span 2 / span 2; }
  .smolitux-grid-item[data-xs="3"] { grid-column: span 3 / span 3; }
  .smolitux-grid-item[data-xs="4"] { grid-column: span 4 / span 4; }
  .smolitux-grid-item[data-xs="5"] { grid-column: span 5 / span 5; }
  .smolitux-grid-item[data-xs="6"] { grid-column: span 6 / span 6; }
  .smolitux-grid-item[data-xs="7"] { grid-column: span 7 / span 7; }
  .smolitux-grid-item[data-xs="8"] { grid-column: span 8 / span 8; }
  .smolitux-grid-item[data-xs="9"] { grid-column: span 9 / span 9; }
  .smolitux-grid-item[data-xs="10"] { grid-column: span 10 / span 10; }
  .smolitux-grid-item[data-xs="11"] { grid-column: span 11 / span 11; }
  .smolitux-grid-item[data-xs="12"] { grid-column: span 12 / span 12; }
  .smolitux-grid-item[data-xs="auto"] { grid-column: auto; }
}

@media (min-width: 577px) and (max-width: 768px) {
  .smolitux-grid-item[data-sm="1"] { grid-column: span 1 / span 1; }
  .smolitux-grid-item[data-sm="2"] { grid-column: span 2 / span 2; }
  .smolitux-grid-item[data-sm="3"] { grid-column: span 3 / span 3; }
  .smolitux-grid-item[data-sm="4"] { grid-column: span 4 / span 4; }
  .smolitux-grid-item[data-sm="5"] { grid-column: span 5 / span 5; }
  .smolitux-grid-item[data-sm="6"] { grid-column: span 6 / span 6; }
  .smolitux-grid-item[data-sm="7"] { grid-column: span 7 / span 7; }
  .smolitux-grid-item[data-sm="8"] { grid-column: span 8 / span 8; }
  .smolitux-grid-item[data-sm="9"] { grid-column: span 9 / span 9; }
  .smolitux-grid-item[data-sm="10"] { grid-column: span 10 / span 10; }
  .smolitux-grid-item[data-sm="11"] { grid-column: span 11 / span 11; }
  .smolitux-grid-item[data-sm="12"] { grid-column: span 12 / span 12; }
  .smolitux-grid-item[data-sm="auto"] { grid-column: auto; }
}

@media (min-width: 769px) and (max-width: 992px) {
  .smolitux-grid-item[data-md="1"] { grid-column: span 1 / span 1; }
  .smolitux-grid-item[data-md="2"] { grid-column: span 2 / span 2; }
  .smolitux-grid-item[data-md="3"] { grid-column: span 3 / span 3; }
  .smolitux-grid-item[data-md="4"] { grid-column: span 4 / span 4; }
  .smolitux-grid-item[data-md="5"] { grid-column: span 5 / span 5; }
  .smolitux-grid-item[data-md="6"] { grid-column: span 6 / span 6; }
  .smolitux-grid-item[data-md="7"] { grid-column: span 7 / span 7; }
  .smolitux-grid-item[data-md="8"] { grid-column: span 8 / span 8; }
  .smolitux-grid-item[data-md="9"] { grid-column: span 9 / span 9; }
  .smolitux-grid-item[data-md="10"] { grid-column: span 10 / span 10; }
  .smolitux-grid-item[data-md="11"] { grid-column: span 11 / span 11; }
  .smolitux-grid-item[data-md="12"] { grid-column: span 12 / span 12; }
  .smolitux-grid-item[data-md="auto"] { grid-column: auto; }
}

@media (min-width: 993px) and (max-width: 1200px) {
  .smolitux-grid-item[data-lg="1"] { grid-column: span 1 / span 1; }
  .smolitux-grid-item[data-lg="2"] { grid-column: span 2 / span 2; }
  .smolitux-grid-item[data-lg="3"] { grid-column: span 3 / span 3; }
  .smolitux-grid-item[data-lg="4"] { grid-column: span 4 / span 4; }
  .smolitux-grid-item[data-lg="5"] { grid-column: span 5 / span 5; }
  .smolitux-grid-item[data-lg="6"] { grid-column: span 6 / span 6; }
  .smolitux-grid-item[data-lg="7"] { grid-column: span 7 / span 7; }
  .smolitux-grid-item[data-lg="8"] { grid-column: span 8 / span 8; }
  .smolitux-grid-item[data-lg="9"] { grid-column: span 9 / span 9; }
  .smolitux-grid-item[data-lg="10"] { grid-column: span 10 / span 10; }
  .smolitux-grid-item[data-lg="11"] { grid-column: span 11 / span 11; }
  .smolitux-grid-item[data-lg="12"] { grid-column: span 12 / span 12; }
  .smolitux-grid-item[data-lg="auto"] { grid-column: auto; }
}

@media (min-width: 1201px) {
  .smolitux-grid-item[data-xl="1"] { grid-column: span 1 / span 1; }
  .smolitux-grid-item[data-xl="2"] { grid-column: span 2 / span 2; }
  .smolitux-grid-item[data-xl="3"] { grid-column: span 3 / span 3; }
  .smolitux-grid-item[data-xl="4"] { grid-column: span 4 / span 4; }
  .smolitux-grid-item[data-xl="5"] { grid-column: span 5 / span 5; }
  .smolitux-grid-item[data-xl="6"] { grid-column: span 6 / span 6; }
  .smolitux-grid-item[data-xl="7"] { grid-column: span 7 / span 7; }
  .smolitux-grid-item[data-xl="8"] { grid-column: span 8 / span 8; }
  .smolitux-grid-item[data-xl="9"] { grid-column: span 9 / span 9; }
  .smolitux-grid-item[data-xl="10"] { grid-column: span 10 / span 10; }
  .smolitux-grid-item[data-xl="11"] { grid-column: span 11 / span 11; }
  .smolitux-grid-item[data-xl="12"] { grid-column: span 12 / span 12; }
  .smolitux-grid-item[data-xl="auto"] { grid-column: auto; }
}

/* Ausrichtung */
.smolitux-grid.align-items-start {
  align-items: start;
}

.smolitux-grid.align-items-center {
  align-items: center;
}

.smolitux-grid.align-items-end {
  align-items: end;
}

.smolitux-grid.align-items-stretch {
  align-items: stretch;
}

.smolitux-grid.justify-content-start {
  justify-content: start;
}

.smolitux-grid.justify-content-center {
  justify-content: center;
}

.smolitux-grid.justify-content-end {
  justify-content: end;
}

.smolitux-grid.justify-content-space-between {
  justify-content: space-between;
}

.smolitux-grid.justify-content-space-around {
  justify-content: space-around;
}

.smolitux-grid.justify-content-space-evenly {
  justify-content: space-evenly;
}