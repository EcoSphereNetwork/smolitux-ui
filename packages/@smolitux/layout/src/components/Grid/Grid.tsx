// packages/@smolitux/layout/src/components/Grid/Grid.tsx
import React, { forwardRef } from 'react';

export interface GridProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Container-Modus aktivieren (für Grid-Container) */
  container?: boolean;
  /** Item-Modus aktivieren (für Grid-Items) */
  item?: boolean;
  /** Abstand zwischen Grid-Items */
  spacing?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 8 | 10 | 12;
  /** Horizontaler Abstand zwischen Grid-Items */
  columnSpacing?: GridProps['spacing'];
  /** Vertikaler Abstand zwischen Grid-Items */
  rowSpacing?: GridProps['spacing'];
  /** Ausrichtung der Items entlang der Hauptachse */
  justifyContent?: 'flex-start' | 'center' | 'flex-end' | 'space-between' | 'space-around' | 'space-evenly';
  /** Ausrichtung der Items entlang der Kreuzachse */
  alignItems?: 'flex-start' | 'center' | 'flex-end' | 'stretch' | 'baseline';
  /** Flex-Direction der Items */
  direction?: 'row' | 'row-reverse' | 'column' | 'column-reverse';
  /** Flex-Wrap-Verhalten */
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
  /** Anzahl der Spalten für xs-Bildschirme (0px+) */
  xs?: number | 'auto' | boolean;
  /** Anzahl der Spalten für sm-Bildschirme (640px+) */
  sm?: number | 'auto' | boolean;
  /** Anzahl der Spalten für md-Bildschirme (768px+) */
  md?: number | 'auto' | boolean;
  /** Anzahl der Spalten für lg-Bildschirme (1024px+) */
  lg?: number | 'auto' | boolean;
  /** Anzahl der Spalten für xl-Bildschirme (1280px+) */
  xl?: number | 'auto' | boolean;
  /** Anzahl der Spalten für 2xl-Bildschirme (1536px+) */
  xxl?: number | 'auto' | boolean;
}

const GRID_COLS = 12;

/**
 * Grid-Komponente für responsive Layouts
 * 
 * @example
 * ```tsx
 * <Grid container spacing={4}>
 *   <Grid item xs={12} md={6}>
 *     <Card>Inhalt 1</Card>
 *   </Grid>
 *   <Grid item xs={12} md={6}>
 *     <Card>Inhalt 2</Card>
 *   </Grid>
 * </Grid>
 * ```
 */
export const Grid = forwardRef<HTMLDivElement, GridProps>(({
  container = false,
  item = false,
  spacing = 0,
  columnSpacing,
  rowSpacing,
  justifyContent,
  alignItems,
  direction,
  wrap = 'wrap',
  xs,
  sm,
  md,
  lg,
  xl,
  xxl,
  className = '',
  children,
  ...rest
}, ref) => {
  // Spacing-Klassen umrechnen
  const getSpacing = (value: number | undefined) => {
    if (value === undefined) return '';
    return `gap-${value}`;
  };

  // Responsive Col-Span-Klassen generieren
  const getColumnClass = (breakpoint: string, value: number | 'auto' | boolean | undefined) => {
    if (value === undefined) return '';
    if (value === 'auto') return `${breakpoint}:col-auto`;
    if (value === true) return `${breakpoint}:col`;
    if (typeof value === 'number') return `${breakpoint}:col-span-${value}`;
    return '';
  };

  // CSS-Klassen zusammenstellen
  const classes = [
    // Container-Klassen
    container ? 'grid' : '',
    container && spacing !== undefined ? getSpacing(spacing) : '',
    container && columnSpacing !== undefined ? `gap-x-${columnSpacing}` : '',
    container && rowSpacing !== undefined ? `gap-y-${rowSpacing}` : '',
    
    // Item-Klassen
    item ? 'col-span-full' : '', // Default auf volle Breite
    item ? getColumnClass('', xs) : '',
    item ? getColumnClass('sm', sm) : '',
    item ? getColumnClass('md', md) : '',
    item ? getColumnClass('lg', lg) : '',
    item ? getColumnClass('xl', xl) : '',
    item ? getColumnClass('2xl', xxl) : '',
    
    // Flexbox-Eigenschaften
    justifyContent ? `justify-${justifyContent}` : '',
    alignItems ? `items-${alignItems}` : '',
    direction ? `flex-${direction}` : '',
    wrap ? `flex-${wrap}` : '',
    
    // Benutzerdefinierte Klassen
    className
  ].filter(Boolean).join(' ');
  
  // Container-spezifische Logik
  let gridTemplateColumns = '';
  if (container) {
    gridTemplateColumns = 'grid-cols-12'; // Default 12-Spalten-Grid
  }
  
  return (
    <div
      ref={ref}
      className={`${classes} ${container ? gridTemplateColumns : ''}`}
      {...rest}
    >
      {children}
    </div>
  );
});

Grid.displayName = 'Grid';

export default Grid;
