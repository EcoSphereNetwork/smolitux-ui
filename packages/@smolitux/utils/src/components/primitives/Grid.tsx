import React, { forwardRef } from 'react';
import { Box, BoxProps } from './Box';

export interface GridProps extends BoxProps {
  /** Number of columns */
  columns?: number | string;
  /** Number of rows */
  rows?: number | string;
  /** Gap between items */
  gap?: number | string;
  /** Column gap */
  columnGap?: number | string;
  /** Row gap */
  rowGap?: number | string;
  /** Template areas */
  areas?: string;
  /** Auto columns */
  autoColumns?: string;
  /** Auto rows */
  autoRows?: string;
  /** Auto flow */
  autoFlow?: 'row' | 'column' | 'dense' | 'row dense' | 'column dense';
}

/**
 * Grid is a layout component that uses CSS Grid to arrange its children.
 * It extends the Box component with grid-specific properties.
 */
export const Grid = forwardRef<HTMLDivElement, GridProps>(
  (
    {
      columns,
      rows,
      gap,
      columnGap,
      rowGap,
      areas,
      autoColumns,
      autoRows,
      autoFlow,
      style,
      className = '',
      ...rest
    },
    ref
  ) => {
    const gridStyle: React.CSSProperties = {
      display: 'grid',
      gridTemplateColumns: typeof columns === 'number' ? `repeat(${columns}, 1fr)` : columns,
      gridTemplateRows: typeof rows === 'number' ? `repeat(${rows}, 1fr)` : rows,
      gap,
      columnGap,
      rowGap,
      gridTemplateAreas: areas,
      gridAutoColumns: autoColumns,
      gridAutoRows: autoRows,
      gridAutoFlow: autoFlow,
      ...style,
    };

    return <Box ref={ref} className={className} style={gridStyle} {...rest} />;
  }
);

Grid.displayName = 'Grid';