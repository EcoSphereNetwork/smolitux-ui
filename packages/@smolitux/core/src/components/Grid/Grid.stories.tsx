import type { Meta, StoryObj } from '@storybook/react';
import { Grid } from './Grid';

const meta: Meta<typeof Grid> = {
  title: 'Components/Grid',
  component: Grid,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Grid',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom Grid',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive Grid',
    onClick: () => alert('Clicked!'),
  },
};
