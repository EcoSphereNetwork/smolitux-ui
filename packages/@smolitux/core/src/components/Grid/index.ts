import { default as BaseGrid, Grid as OriginalGrid, type GridProps } from './Grid';
import { default as GridA11y, type GridA11yProps } from './Grid.a11y';

// Erweitere Grid um die A11y-Komponente
const Grid = Object.assign(OriginalGrid, {
  A11y: GridA11y
});

// Exportiere Komponenten und Typen
export { Grid };
export type { GridProps, GridA11yProps };

// Fuer Abwaertskompatibilitaet mit dem bestehenden Export
export default Grid;