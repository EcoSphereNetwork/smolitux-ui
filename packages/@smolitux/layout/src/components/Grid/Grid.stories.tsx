import type { Meta, StoryObj } from '@storybook/react';
import Grid from './Grid';

const meta: Meta<typeof Grid> = {
  title: 'Layout/Grid',
  component: Grid,
  parameters: {
    layout: 'padded',
  },
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof Grid>;

const Box = ({ children }: { children: React.ReactNode }) => (
  <div style={{ background: '#e5e7eb', padding: '10px', border: '1px solid #d1d5db' }}>
    {children}
  </div>
);

export const Default: Story = {
  render: () => (
    <Grid gap="sm">
      <Grid.Item xs={12} md={6}>
        <Box>Item 1</Box>
      </Grid.Item>
      <Grid.Item xs={12} md={6}>
        <Box>Item 2</Box>
      </Grid.Item>
    </Grid>
  ),
};

export const Responsive: Story = {
  render: () => (
    <Grid gap={{ sm: 'xs', md: 'md' }} cols={{ sm: 2, md: 4 }} justify={{ md: 'center' }}>
      <Grid.Item xs={12} sm={6} md={2}>
        <Box>One</Box>
      </Grid.Item>
      <Grid.Item xs={12} sm={6} md={2}>
        <Box>Two</Box>
      </Grid.Item>
    </Grid>
  ),
};
