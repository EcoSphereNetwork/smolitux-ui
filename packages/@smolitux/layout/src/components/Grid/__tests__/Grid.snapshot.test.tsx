import React from 'react';
import { render } from '@testing-library/react';
import Grid from '../Grid';

describe('Grid Snapshots', () => {
  it('renders container with two items', () => {
    const { asFragment } = render(
      <Grid gap="md">
        <Grid.Item xs={6}>One</Grid.Item>
        <Grid.Item xs={6}>Two</Grid.Item>
      </Grid>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
