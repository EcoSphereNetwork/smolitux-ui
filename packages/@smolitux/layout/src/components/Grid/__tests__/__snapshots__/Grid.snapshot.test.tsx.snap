// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Snapshots renders container with two items 1`] = `
<DocumentFragment>
  <div
    class="grid grid-cols-12 gap-md justify-start items-stretch"
  >
    <div
      class="col-span-6"
    >
      One
    </div>
    <div
      class="col-span-6"
    >
      Two
    </div>
  </div>
</DocumentFragment>
`;
